src/pages/api/searchOperators.ts

```diff
@@ -0,0 +1,39 @@
+import { NextApiResponse } from 'next';
+import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+import { SEARCH_OPERATOR_ATTRIBUTE } from '../../constants';
+import { redirectTo, redirectToError } from './apiUtils';
+import { updateSessionAttribute } from '../../utils/sessions';
+
+export interface SearchOperators {
+    searchResults: string[];
+}
+
+export interface SearchOperatorsWithErrors extends SearchOperators {
+    errors: ErrorInfo[];
+}
+
+export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
+    const errors: ErrorInfo[] = [];
+
+    try {
+        const { searchText } = req.body;
+
+        if (searchText.length < 4) {
+            errors.push({
+                errorMessage: 'Search requires a minimum of three characters',
+                id: 'searchText',
+            });
+        }
+
+        if (errors.length > 0) {
+            updateSessionAttribute(req, SEARCH_OPERATOR_ATTRIBUTE, { errors, searchResults: [] });
+            redirectTo(res, '/searchOperators');
+        }
+
+        updateSessionAttribute(req, SEARCH_OPERATOR_ATTRIBUTE, { errors: [], searchResults: [] });
```

Comment:
Do we need searchResults in the session if its always going to be an empty array?

Reply:
removed and if needed for future work can be added